src/main/java/org/commcare/xml/IndexedFixtureXmlParser.java

```diff
@@ -0,0 +1,120 @@
+package org.commcare.xml;
+
+import org.commcare.cases.instance.FixtureIndexSchema;
+import org.commcare.cases.model.StorageIndexedTreeElementModel;
+import org.commcare.core.interfaces.UserSandbox;
+import org.commcare.data.xml.TransactionParser;
+import org.javarosa.core.model.instance.TreeElement;
+import org.javarosa.core.services.storage.IStorageUtilityIndexed;
+import org.javarosa.core.services.storage.StorageFullException;
+import org.javarosa.xml.TreeElementParser;
+import org.javarosa.xml.util.InvalidStructureException;
+import org.javarosa.xml.util.UnfullfilledRequirementsException;
+import org.kxml2.io.KXmlParser;
+import org.xmlpull.v1.XmlPullParserException;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Creates a table for the indexed fixture and parses each element into a
+ * StorageIndexedTreeElementModel and stores that as a table row.
+ *
+ * Also stores base and child names associated with fixture in another database.
+ * For example, if we have a fixture referenced by
+ * instance('product-list')/products/product/... then we need to associate
+ * ('product-list', 'products', 'product') to be able to reconstruct the
+ * fixture instance
+ *
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+public class IndexedFixtureXmlParser extends TransactionParser<StorageIndexedTreeElementModel> {
+
+    private final Set<String> indices;
+    private final Set<String> columnIndices;
+    private final UserSandbox sandbox;
+    private final String fixtureName;
+    private IStorageUtilityIndexed<StorageIndexedTreeElementModel> indexedFixtureStorage;
+
+    public IndexedFixtureXmlParser(KXmlParser parser, String fixtureName,
+                                   FixtureIndexSchema schema, UserSandbox sandbox) {
+        super(parser);
+        this.sandbox = sandbox;
+        this.fixtureName = fixtureName;
+
+        if (schema == null) {
```

Comment:
Flagging that it's going to be hard for users to identify when they've made the mistake of sending down the index data _after_ the fixture data, since it will work, it'll just be super slow.

We should possibly make "including a schema element _after_ a fixture element with the same name" an error, even if having a fixture element with no schema is not.